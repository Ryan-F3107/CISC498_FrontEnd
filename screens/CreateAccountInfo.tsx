import React from 'react';
import { Text, View, TouchableOpacity, Platform, RefreshControlBase } from 'react-native';
import { TextInput } from 'react-native-paper';
import styles from '../styling/styles';

class CreateAccountInfo extends React.Component {

    constructor(props) {
        super(props);
        const initalState = {
            email: '',
            password: '',
            confirmPassword: '',
            errorEmail: '',
            errorPassword: '',
            errorConfirmPassword: '',
        }
        this.state = initalState;
    }

    // Verify that all the required fields are filled in
    checkForm() {
        let decision = false;

        if (this.state.errorEmail == "Invalid" || this.state.email == "" || this.state.password.length < 8
            || this.state.confirmPassword != this.state.password) {
            decision = true
        }
        else {
            decision = false
        }
        return decision

    }

    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.title}> Create an Account </Text>

                {/*Email Address*/}
                <TextInput
                    style={styles.textInput}
                    label="EMAIL"
                    mode="outlined"
                    autoCapitalize='none'
                    placeholder="myemail@domain.com"
                    theme={{ colors: { primary: '#0a0540' } }}
                    onChangeText={email => this.setState(() => ({ email: email }))}
                    value={this.state.email}
                    onBlur={() => { // If the field is left blank, or has an invalid email address, show an error message
                        if (this.state.email == "") {
                            this.setState(() => ({ errorEmail: "Required" }));
                        } else if (this.state.email.length <= 5 || !this.state.email.includes('@') && !this.state.email.includes('.')) {
                            this.setState(() => ({ errorEmail: "Invalid" }));
                        }
                    }}
                    onFocus={() => { // When the field is tapped, remove the error message
                        this.setState(() => ({ errorEmail: "" }));
                    }}
                />
                <Text style={styles.errorMessage}>{this.state.errorEmail}</Text>

                <View style={{ padding: 20, borderBottomWidth: 1, borderBottomColor: 'grey' }}></View>

                {/*Password*/}
                <TextInput
                    style={styles.textInputPassword}
                    label="PASSWORD"
                    mode="outlined"
                    autoCapitalize='none'
                    secureTextEntry={true}
                    theme={{ colors: { primary: '#0a0540' } }}
                    onChangeText={password => this.setState(() => ({ password: password }))}
                    value={this.state.password}
                    onBlur={() => { // If the field is left blank, or has an invalid password, show an error message 
                        if (this.state.password == "") {
                            this.setState(() => ({ errorPassword: "Required" }));
                            this.setState(() => ({ confirmPassword: '' }));
                        } else if (this.state.password.length < 8) {
                            this.setState(() => ({ errorPassword: "Must be at least 8 characters long" }));
                        }
                    }}
                    onFocus={() => { // When the field is tapped, remove the error message
                        this.setState(() => ({ errorPassword: "", confirmPassword: "" }));
                    }}
                />
                <Text style={styles.errorMessage}>{this.state.errorPassword}</Text>

                {/*Re-enter Password*/}
                <TextInput
                    style={styles.textInputConfirmPassword}
                    label="CONFIRM PASSWORD"
                    mode="outlined"
                    autoCapitalize='none'
                    secureTextEntry={true}
                    theme={{ colors: { primary: '#0a0540' } }}
                    disabled={this.state.password.length < 8}
                    onChangeText={confirmPassword => this.setState(() => ({ confirmPassword: confirmPassword }))}
                    value={this.state.confirmPassword}
                    onBlur={() => { // If the field is left blank or if the password and re-entered password don't match, show an error message 
                        if (this.state.confirmPassword == "") {
                            this.setState(() => ({ errorConfirmPassword: "Required" }));
                        } else if (this.state.password != this.state.confirmPassword) {
                            this.setState(() => ({ errorConfirmPassword: "Paswords do not match" }));
                        }
                    }}
                    onFocus={() => { // When the field is tapped, remove the error message
                        this.setState(() => ({ errorConfirmPassword: "" }));
                    }}
                />
                <Text style={styles.errorMessage}>{this.state.errorConfirmPassword}</Text>

                {/*Button - send all the entered info*/}
                <View style={{
                    position: (Platform.OS === 'ios') ? "absolute" : "relative",
                    bottom: (Platform.OS === 'ios') ? 210 : -80,
                    alignSelf: 'center'
                }}
                >
                    <TouchableOpacity
                        style={styles.button}
                        onPress={() => {

                            if (this.props.route.params.accountType == "customer") {
                                this.props.navigation.navigate('Terms_Conditions',
                                    {
                                        accountType: this.props.route.params.accountType,
                                        firstName: this.props.route.params.firstName,
                                        lastName: this.props.route.params.lastName,
                                        phoneNum: this.props.route.params.phoneNum,
                                        email: this.state.email,
                                        password: this.state.password
                                    });

                            } else if (this.props.route.params.accountType == "business") {
                                this.props.navigation.navigate('Terms_Conditions',
                                    {
                                        accountType: this.props.route.params.accountType,
                                        businessName: this.props.route.params.businessName,
                                        phoneNum: this.props.route.params.phoneNum,
                                        street: this.props.route.params.street,
                                        city: this.props.route.params.city,
                                        province: this.props.route.params.province,
                                        postalCode: this.props.route.params.postalCode,
                                        capacity: this.props.route.params.capacity,
                                        email: this.state.email,
                                        password: this.state.password
                                    });
                            }
                        }}
                        disabled={this.checkForm()}
                    >
                        <Text style={{ color: '#fafafa', alignSelf: 'center' }}>Sign Up</Text>
                    </TouchableOpacity>
                </View>


            </View>

        )
    }
}

export default CreateAccountInfo;