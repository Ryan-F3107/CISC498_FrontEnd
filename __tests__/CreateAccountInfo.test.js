import React from 'react';
import renderer from 'react-test-renderer';
import CreateAccountInfo from '../screens/CreateAccountInfo';

describe("<CreateAccountInfo />", () => {
    it("has 1 child", () => {
        const tree = renderer.create(<CreateAccountInfo />).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('renders correctly', () => {
        renderer.create(<CreateAccountInfo />);
    });

    it('CreateAccountInfo testing with snapshot', () => {
        const tree = renderer.create(<CreateAccountInfo />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});