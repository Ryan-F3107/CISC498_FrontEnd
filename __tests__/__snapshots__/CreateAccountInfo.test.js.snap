// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateAccountInfo /> CreateAccountInfo testing with snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fafafa",
        "flex": 1,
        "padding": 70,
        "paddingBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 28,
          "marginBottom": 30,
          "paddingBottom": 20,
          "paddingTop": 50,
        }
      }
    >
       Create an Account 
    </Text>
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "paddingTop": 12,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#fafafa",
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 64,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  EMAIL
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#0a0540",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  EMAIL
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  EMAIL
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                selectionColor="#0a0540"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "red",
              "fontSize": 11,
              "paddingBottom": 0,
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "padding": 20,
            }
          }
        />
        <View
          style={
            Object {
              "paddingTop": 16,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#fafafa",
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 64,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  PASSWORD
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#0a0540",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  PASSWORD
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  PASSWORD
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                secureTextEntry={true}
                selectionColor="#0a0540"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "red",
              "fontSize": 11,
              "paddingBottom": 0,
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "paddingTop": 12,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#fafafa",
                    "borderColor": "rgba(0, 0, 0, 0.26)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 64,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#fafafa",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  CONFIRM PASSWORD
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  CONFIRM PASSWORD
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.26)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  CONFIRM PASSWORD
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                editable={false}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.26)"
                rejectResponderTermination={true}
                secureTextEntry={true}
                selectionColor="rgba(0, 0, 0, 0.54)"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "red",
              "fontSize": 11,
              "paddingBottom": 0,
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "marginBottom": 130,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#0a0540",
              "borderRadius": 25,
              "opacity": 1,
              "padding": 15,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#fafafa",
              }
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
